netx/net/sourceforge/jnlp/runtime/JNLPProxySelector.java

```diff
@@ -393,6 +471,115 @@ private boolean isLocalHost(String host) {
+         try {
+             proxyEnable = WinRegistry.readDword (
+                 PROXY_DWORD_REGISTRY_KEY,
+                 PROXY_ENABLE_REGISTRY_VAL
+             );
+
+           OutputController.getLogger().log("ProxyEnable set to " + proxyEnable);
+         }
+         catch (Exception e) {
+            OutputController.getLogger().log("Error reading from Windows registry or Operating System is not Windows based.");
+            return new ArrayList<>();
+         } */
+        try {
+
+//             if (proxyEnable.equals("0x1")) {
+            proxyServer = WinRegistry.readString (
+                    WinRegistry.HKEY_CURRENT_USER,
+                    PROXY_REGISTRY_KEY,
+                    PROXY_SERVER_REGISTRY_VAL, 0);
+
+            if ((proxyServer.startsWith("ftp=")) || (proxyServer.startsWith("http=")) || (proxyServer.startsWith("https=")) || (proxyServer.startsWith("socks="))) {
+                proxyServer = proxyServer.replaceAll("\\.*.*=" , "").replaceAll(";.*" , "");
+            }
+            else {
+                proxyServer = proxyServer;
+            }
+            setOverrideHosts(
+                    WinRegistry.readString(
+                            WinRegistry.HKEY_CURRENT_USER,
+                            PROXY_REGISTRY_KEY,
+                            PROXY_SERVER_OVERRIDE_VAL, 0)
+            );
+
+//                 OutputController.getLogger().log("ProxyServer set to " + proxyServer);
+//                 OutputController.getLogger().log("ProxyOverrides set to " + getOverrideHosts());
+
+
+        }
+        catch (Exception e) {
+//            OutputController.getLogger().log("WARNING: Found ProxyEnable set to 1, but could not read value of ProxyServer");
+        }
+//         OutputController.getLogger().log("Found proxy server '" + proxyServer + "' in the Windows Registry");
+        List<Proxy> proxies = new ArrayList<>();
+        String host = "";
+        host = proxyServer.split(":")[0];
+        int port = Integer.valueOf(proxyServer.split(":")[1]);
+
+        if (!host.isEmpty()) {
+            try {
+                SocketAddress sa = new InetSocketAddress(host, port);
+                proxies.add(new Proxy(Type.HTTP, sa));
+            }
+            catch (Exception e) {
+                OutputController.getLogger().log("Could not parse proxy settings: " + e.getMessage());
+            }
+        }
+
+        if (checkProxyOverride(getOverrideHosts(), uri)) {
+            proxies = new ArrayList<>();
+            proxies.add(Proxy.NO_PROXY);
+        }
+
+        return proxies;
+
+    } // getFromWindowsRegistry
+
+    public static Boolean checkProxyOverride(String proxyOverrides, URI uri) {
+
+        Boolean overrideFound = false;
+        if (proxyOverrides != null) {
+            String[] overrides = proxyOverrides.split(";");
+            for (String override : overrides) {
+//            OutputController.getLogger().log("Proxy override check: Comparing '" +
```

Comment:
I'm in favour to drop all those comment, or to enable it. Logging to most verbose log is good thing to do.